evm_interpreter/src/lib.rs

```diff
@@ -181,34 +181,22 @@ impl<S: SystemTypes> BitMap<S> {
 
 /// Analyzs bytecode to build a jump map.
 fn analyze<S: SystemTypes>(code: &[u8], system: &mut System<S>) -> Result<BitMap<S>, ()> {
+    use self::opcodes as opcode;
+
     let code_len = code.len();
     let mut jumps = BitMap::<S>::allocate_for_bit_capacity(code_len, system);
-    let mut code_iter = code.iter().copied().enumerate();
 
-    use self::opcodes as opcode;
-
-    while let Some((offset, opcode)) = code_iter.next() {
-        if opcode::JUMPDEST == opcode {
-            // SAFETY: jumps are max length of the code
-            unsafe { jumps.set_bit_on_unchecked(offset) }
-            // step by 1 is automatic
+    let mut i = 0;
+    while i < code_len {
```

Comment:
`.enumerate()` should produce the same or better code than while. Though in this case there probably is no bounds check even with while.

Reply:
My bad, didn't read further.